test/WebJobs.Script.Tests/WorkerFunctionMetadataProviderTests.cs

```diff
@@ -85,7 +85,7 @@ public void ValidateBindings_NoTriggerBinding_Throws()
         [Theory]
         [InlineData(null)]
         [InlineData("")]
-        [InlineData("_binding")]
+        [InlineData("__binding__")]
```

Comment:
Your regex disallows __ anywhere, so perhaps you should add cases:

__binding
binding__binding
binding__